### ISSUE TITLE
Not working with slurm 20.02

### ISSUE THREAD

**Issue by jlbl** [mercanca/spart#6]
We've got a test installation of slurm 20.02.0.  After compiling via `gcc -lslurm spart.c -o spart`, the resulting binary throws this error:
```
spart: symbol lookup error: /usr/lib64/slurm/accounting_storage_slurmdbd.so: undefined symbol: running_in_slurmctld
```

Thanks.

**Comment by ssgituser** [comment]
Same issue here on Centos 7.7 and Slurm 20.02

spart: symbol lookup error: /usr/lib64/slurm/accounting_storage_slurmdbd.so: undefined symbol: running_in_slurmctld

**Comment by mercanca** [comment]
I don't have a slurm 20.02. But would you try to compile as below:
gcc -lslurm -lslurmdb spart.c -o spart
If this will not work, I will try to install slurm 20.02 to see what is wrong about it.

**Comment by jlbl** [comment]
```
$ gcc -lslurm -lslurmdb spart.c -o spart
/usr/bin/ld: cannot find -lslurmdb
collect2: error: ld returned 1 exit status
```

Also on CentOS 7, btw.

**Comment by mercanca** [comment]
Is your slurm instalation contains mariadb server. Please check your rpms using this command:

rpm -ql slurm-slurmdbd | grep accounting_storage_mysql.so

It should show the location of this file.

**Comment by jlbl** [comment]
It's in the standard location:
```
$ rpm -ql slurm-slurmdbd | grep accounting_storage_mysql.so
/usr/lib64/slurm/accounting_storage_mysql.so
```
Also, based on the original error message, it can find the library - it's just missing the symbol `running_in_slurmctld`.

**Comment by mercanca** [comment]
I prepare a quick fix to this problem, but the spart will work with reduced feature set.
At slurm 20.02, the connection to the slurm database is not working. I am looking for a solution to this problem. But, until we find a better solution, the only thing I can do, is not using slurm database at slurm 20.02.

**Comment by ssgituser** [comment]
Thank you for the quick fix unfortunately not using the Slurm database is not an option. Do you think a fix will be available in the future for using 20.02 with the DB and this spart tool?

**Comment by mercanca** [comment]
I report a bug to slurm, but I don't have a support contract. I don't know how they handle this bug.

**Comment by ssgituser** [comment]
Unfortunately from past experience without a support contract they will probably not look at the bug.

--
___________________________________________________
Sajesh Singh
Manager, Systems and Scientific Computing
American Museum of Natural History
200 Central Park West
New York, NY 10024

(O) (212) 313-7263
(C) (917) 763-9038
(E) ssingh@amnh.org
________________________________

**Comment by mercanca** [comment]
The Slurm support replied as this problem will be fixed at slurm 20.02.2.

**Comment by mercanca** [comment]
I uploaded a new version of the spart (v1.2.1) for the slurm 20.02.2. The spart works as it should be, now.
